test/cpp/naming/resolver_component_test.cc

```diff
@@ -0,0 +1,338 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <grpc/grpc.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/host_port.h>
+#include <grpc/support/log.h>
+#include <grpc/support/string_util.h>
+#include <grpc/support/sync.h>
+#include <grpc/support/time.h>
+#include <string.h>
+
+#include <gflags/gflags.h>
+#include <gmock/gmock.h>
+#include <vector>
+
+#include "test/cpp/util/subprocess.h"
+#include "test/cpp/util/test_config.h"
+
+extern "C" {
+#include "src/core/ext/filters/client_channel/client_channel.h"
+#include "src/core/ext/filters/client_channel/resolver.h"
+#include "src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.h"
+#include "src/core/ext/filters/client_channel/resolver_registry.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/iomgr/combiner.h"
+#include "src/core/lib/iomgr/executor.h"
+#include "src/core/lib/iomgr/iomgr.h"
+#include "src/core/lib/iomgr/resolve_address.h"
+#include "src/core/lib/iomgr/sockaddr_utils.h"
+#include "src/core/lib/support/env.h"
+#include "src/core/lib/support/string.h"
+#include "test/core/util/port.h"
+#include "test/core/util/test_config.h"
+}
+
+using std::vector;
+using grpc::SubProcess;
+using testing::UnorderedElementsAreArray;
+
+// Hack copied from "test/cpp/end2end/server_crash_test_client.cc"!
+// In some distros, gflags is in the namespace google, and in some others,
+// in gflags. This hack is enabling us to find both.
+namespace google {}
+namespace gflags {}
+using namespace google;
+using namespace gflags;
+
+DEFINE_string(target_name, "", "Target name to resolve.");
+DEFINE_string(expected_addrs, "",
+              "Comma-separated list of expected "
+              "'<ip0:port0>,<is_balancer0>;<ip1:port1>,<is_balancer1>;...' "
+              "addresses of "
+              "backend and/or balancers. 'is_balancer' should be bool, i.e. "
+              "true or false.");
+DEFINE_string(expected_chosen_service_config, "",
+              "Expected service config json string that gets chosen (no "
+              "whitespace). Empty for none.");
+DEFINE_string(
+    local_dns_server_address, "",
+    "Optional. This address is placed as the uri authority if present.");
+DEFINE_string(expected_lb_policy, "",
+              "Expected lb policy name that appears in resolver result channel "
+              "arg. Empty for none.");
+
+namespace {
+
+class GrpcLBAddress final {
+ public:
+  GrpcLBAddress(std::string address, bool is_balancer)
+      : is_balancer(is_balancer), address(address) {}
+
+  bool operator==(const GrpcLBAddress &other) const {
+    return this->is_balancer == other.is_balancer &&
+           this->address == other.address;
+  }
+
+  bool operator!=(const GrpcLBAddress &other) const {
+    return !(*this == other);
+  }
+
+  bool is_balancer;
+  std::string address;
+};
+
+bool ConvertStringToBool(std::string bool_str) {
```

Comment:
It looks like a `gpr_is_true()` function was recently added, so we can use that here:

https://github.com/grpc/grpc/blob/master/src/core/lib/support/string.h#L111

Reply:
done